internal/keysharecore/operations.go

```diff
@@ -304,6 +304,62 @@ func (c *Core) GenerateResponse(secrets UserSecrets, accessToken string, commitI
 	return token.SignedString(c.jwtPrivateKey)
 }
 
+// GenerateResponseV2 generates the response of a zero-knowledge proof of the keyshare secret, for a given previous commit and response request.
+func (c *Core) GenerateResponseV2(
+	secrets UserSecrets,
+	accessToken string,
+	commitID uint64,
+	hash gabi.KeyshareCommitmentRequest,
```

Comment:
Maybe you can make the naming more clear here. I first thought that this was just a string or `[]byte` or something, but it appears to be a struct with a list of hashes.